### PR TITLE
IBX-7792: The modal was adapted to the new schema, jQuery dependencies were removed

### PR DESCRIPTION
**Pull request by mateuszdebinski**
Jira: https://issues.ibexa.co/browse/IBX-7792

### CODE REVIEW

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 1]
```diff
@@ -1,57 +1,61 @@
...
-            $(this).removeClass('is-checked');
+(function (global, doc) {
+    const modal = doc.querySelector('#add-translation-modal');
+    const form = modal.querySelector('form[name="add-translation"]');
+    const container = form.querySelector('.ibexa-automated-translation-services-container');
+    const error = container.querySelector('.ibexa-automated-translation-error');
+    const sourceLabel = container.querySelector('.ibexa-field-edit--ezboolean .ibexa-data-source__label');
+
+    const toggleClass = (event) => {
+        const target = event.target;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:9]:
```suggestion
    const toggleClass = ({ target }) => {
```
or 
```suggestion
    const toggleClass = (event) => {
        const { target } = event;
```

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 2]
```diff
@@ -1,57 +1,61 @@
...
+        const input = target.querySelector('input[type="checkbox"]');
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:10]: Are there any ibexa classes on this input? If yes, then it probably should be sth like (at least that's class that SHOULD be on checkbox input ;) )
```suggestion
        const input = target.querySelector('.ibexa-input--checkbox');
```

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 3]
```diff
@@ -1,57 +1,61 @@
...
         }
 
-        let translationAvailable = (typeof sourceLang === 'undefined' || -1 !== $.inArray(sourceLang, mapping[serviceAlias])) && (-1 !== $.inArray(targetLang, mapping[serviceAlias]));
-        if (false === translationAvailable) {
-            $error.removeClass("invisible");
-            if ($container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").hasClass('is-checked')) {
-                $container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").click();
+        const translationAvailable =
+            (typeof sourceLang === 'undefined' || sourceLang.indexOf(languagesMapping[serviceAlias]) !== -1) &&
+            targetLang.indexOf(targetLang, languagesMapping[serviceAlias]) !== -1;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:34]:
```suggestion
        const serviceLang = languagesMapping[serviceAlias];
        const translationAvailable = (typeof sourceLang === 'undefined' || sourceLang.includes(serviceLang)) && targetLang.includes(serviceLang);
```
or sth like this, not sure of the name `serviceLang`, but based on serviceAlias and languagesMapping names I guess it keeps some kind of language

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 4]
```diff
@@ -1,57 +1,61 @@
...
+
                 return false;
             }
         }
+
         return true;
+    };
+
+    form.addEventListener('click', () => error?.classList.add('invisible'));
+    sourceLabel?.addEventListener('click', toggleClass);
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:50]:
```suggestion
    form.addEventListener('click', () => error?.classList.add('invisible'), false);
    sourceLabel?.addEventListener('click', toggleClass, false);
```

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 5]
```diff
@@ -1,57 +1,61 @@
...
+
+    container.querySelectorAll('.ibexa-data-source__label.is-checked').forEach(() => {
+        form.querySelector('.ibexa-btn--create-translation').removeAttribute('disabled');
+    });
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:54]: I think there's something missing in here, as you iterate by `.ibexa-data-source__label.is-checked`, but inside you don't use anything from this iterable array?
I guess it was supposed to remove disabled from button if sth is checked, but in such case it should be enough:
```suggestion
    if (container.querySelector('.ibexa-data-source__label.is-checked')) {
        form.querySelector('.ibexa-btn--create-translation').removeAttribute('disabled');
    };
```

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 6]
```diff
@@ -1,57 +1,61 @@
...
+
+    container.querySelectorAll('.ibexa-dropdown__source select option:checked').forEach(() => {
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:56]: same as above

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 7]
```diff
@@ -1,57 +1,61 @@
...
+        form.querySelector('.ibexa-btn--create-translation').removeAttribute('disabled');
     });
-});
+
+    form.addEventListener('submnit', submitForm);
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:60]:
```suggestion
    form.addEventListener('submnit', submitForm, false);
```

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

**Review by GrabowskiM — APPROVED** [review]

**Review by tischsoic — APPROVED** [review]

**Review by dew326 — APPROVED** [review]

**Review by konradoboza — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 8]
```diff
@@ -1,57 +1,17 @@
...
-        let translationAvailable = (typeof sourceLang === 'undefined' || -1 !== $.inArray(sourceLang, mapping[serviceAlias])) && (-1 !== $.inArray(targetLang, mapping[serviceAlias]));
-        if (false === translationAvailable) {
-            $error.removeClass("invisible");
-            if ($container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").hasClass('is-checked')) {
-                $container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").click();
-                return false;
+((doc) => {
+    const TRANSLATOR_SELECT_SELECTOR = '#add-translation_translatorAlias';
+    const BASE_LANGUAGE_SELECT_SELECTOR = '#add-translation_base_language';
+
```
Comment (dew326) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:4]:
```suggestion
```

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 9]
```diff
@@ -1,57 +1,17 @@
...
+    const translatorSelect = doc.querySelector(TRANSLATOR_SELECT_SELECTOR);
+    const baseLanguageSelect = doc.querySelector(BASE_LANGUAGE_SELECT_SELECTOR);
+    if (baseLanguageSelect && translatorSelect) {
```
Comment (dew326) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:7]:
```suggestion

    if (baseLanguageSelect && translatorSelect) {
```

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 10]
```diff
@@ -1,57 +1,17 @@
...
+        baseLanguageSelect.addEventListener('change', () => {
+            translatorSelect.disabled = !baseLanguageSelect.value;
+
+            const translationSelectWrapper = translatorSelect.closest('.ibexa-dropdown');
+            if (translationSelectWrapper) {
```
Comment (dew326) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:12]:
```suggestion

            if (translationSelectWrapper) {
```

File: src/bundle/Resources/public/admin/js/ibexa-automated-translation.js [hunk 11]
```diff
@@ -1,57 +1,17 @@
...
-            return true;
-        }
-
-        let translationAvailable = (typeof sourceLang === 'undefined' || -1 !== $.inArray(sourceLang, mapping[serviceAlias])) && (-1 !== $.inArray(targetLang, mapping[serviceAlias]));
-        if (false === translationAvailable) {
-            $error.removeClass("invisible");
-            if ($container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").hasClass('is-checked')) {
-                $container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").click();
-                return false;
+((doc) => {
```
Comment (dew326) [inline src/bundle/Resources/public/admin/js/ibexa-automated-translation.js:1]:
```suggestion
((global, doc) => {
```
just for consistency.

**Review by dew326 — APPROVED** [review]

**Comment by micszo** [comment]
Please see: https://recordit.co/zAn4cm11Ux

When opening "Create a new translation" pop-up, after refreshing the page, select translation service dropdown is active with "No language" selected as source language. Which can lead to an error in the next step.

The same occurs when pop-up is opened for the first time for given content item (Dropdown is deactivated when "No language" is selected for the first time).

**Review by micszo — APPROVED** [review]
QA Approved on Ibexa Commerce 4.5.7-dev.

**Pull request merged**